tests/test_new_features.py

```
@@ -0,0 +1,514 @@
+    def test_view_read_error(self, tmp_path, monkeypatch):
+        """Simulate a read error (e.g., permission denied)."""
+        clean_dir = tmp_path / "clean"
+        clean_dir.mkdir()
+        doc = clean_dir / "unreadable.txt"
+        doc.write_text("secret", encoding="utf-8")
+        monkeypatch.setenv("CLEAN_DIR", str(clean_dir))
+        monkeypatch.setenv("RAW_DIR", str(tmp_path / "raw"))
+
+        original_read = Path.read_text
+
+        def raise_on_unreadable(self, *args, **kwargs):
```

Ruff (0.15.5)

[warning] Missing return type annotation for private function `raise_on_unreadable`

(ANN202)